### PR TITLE
Use TestNodePreparer in Density test

### PR DESCRIPTION
**Pull request by gmarek**
Depends on #35129

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35184)

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -4535,6 +4535,7 @@ func NewE2ETestNodePreparer(client clientset.Interface, countToStrategy map[int]
 }
 
 func (p *E2ETestNodePreparer) PrepareNodes() error {
+	glog.Infof("p.client: %p", p.client)
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: remove

**Comment by gmarek** [comment]
@wojtek-t PTAL

File: test/e2e/density.go [hunk 2]
```diff
@@ -461,6 +461,14 @@ var _ = framework.KubeDescribe("Density", func() {
 		name := fmt.Sprintf("[Feature:%s] should allow starting %d pods per node", feature, testArg.podsPerNode)
 		itArg := testArg
 		It(name, func() {
+			nodePreparer := framework.NewE2ETestNodePreparer(
+				f.ClientSet,
+				map[int]testutils.PrepareNodeStrategy{nodeCount: &testutils.TrivialNodePrepareStrategy{}},
+			)
+			err := nodePreparer.PrepareNodes()
+			ExpectNoError(err)
```
Comment (wojtek-t) [inline test/e2e/density.go]: nit: framework.ExpectNoError(nodePreparer.PrepareNodes())

Comment (gmarek) [inline test/e2e/density.go]: Done. @wojtek-t PTAL

**Pull request merged**